Core/include/Acts/EventData/TrackProxy.hpp

```diff
@@ -362,10 +357,31 @@ class TrackProxy {
     return parameters()[eBoundQOverP];
   }
 
+  /// Get the particle hypothesis
+  /// @return the particle hypothesis
+  const ParticleHypothesis& particleHypothesis() const {
+    return component<ParticleHypothesis, hashString("particleHypothesis")>();
+  }
+
+  /// Get the particle hypothesis
+  /// Mutable version
+  /// @return the particle hypothesis
+  template <bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
+  ParticleHypothesis& particleHypothesis() {
+    return component<ParticleHypothesis, hashString("particleHypothesis")>();
+  }
+
+  /// Get the charge of the tack
+  /// @note this depends on the charge hypothesis
+  /// @return The absolute track momentum
+  ActsScalar charge() const {
+    return particleHypothesis().qFromQOP(qOverP());
+  }
+
   /// Get the absolute momentum of the tack
   /// @return The absolute track momentum
   ActsScalar absoluteMomentum() const {
-    return SinglyCharged{}.extractMomentum(qOverP());
+    return particleHypothesis().extractMomentum(qOverP());
```

Comment:
And in this case of temporary access this becomes a bit hairy since we might be recreating these objects quite frequently I suppose.